torch/distributed/_sharded_tensor/api.py

```diff
@@ -335,6 +367,59 @@ def _init_rpc(self):
         # Barrier for all RPCs to finish on all ranks.
         rpc.api._all_gather(None)
 
+    def gather(
+        self,
+        dst: int = 0,
+        out: Optional[torch.Tensor] = None,
+    ) -> None:
+        """
+        Creates a full :class:`Tensor` on rank `dst` by gathering all sharded tensors.
+
+        The API needs to be called on all ranks in SPMD fashion. All ranks should have
+        the same `dst`. `out` should be a full size zero tensor on `dst` and
+        None on all other ranks.
+
+        Args:
+            dst(int): The rank where full tensor is constructed
+            out (:class `torch.Tensor` optional): The output full tensor. Needs to be provided
+            ONLY on `dst`
+        """
+        my_rank = dist.get_rank(self._process_group)
+        full_size = self.metadata().size
+        _validate_output_tensor_for_gather(my_rank, dst, full_size, out)
+        curr_device = torch.device(f"cuda:{my_rank}")
```

Comment:
why we need to use `curr_device` here?

Reply:
As mentioned above, this won't work for ranks > 8